src/15_hello_triangle.rs

```diff
@@ -10,766 +10,739 @@ extern crate glsl_to_spirv;
+    fn create_framebuffers(
+        device: &<back::Backend as Backend>::Device,
+        render_pass: &<back::Backend as Backend>::RenderPass,
+        frame_images: &[(
+            <back::Backend as Backend>::Image,
+            <back::Backend as Backend>::ImageView,
+        )],
+        extent: window::Extent2D,
+    ) -> Vec<<back::Backend as Backend>::Framebuffer> {
+        let mut swapchain_framebuffers: Vec<<back::Backend as Backend>::Framebuffer> = Vec::new();
+
+        for (_, image_view) in frame_images.iter() {
+            swapchain_framebuffers.push(
+                device
+                    .create_framebuffer(
+                        render_pass,
+                        vec![image_view],
+                        image::Extent {
+                            width: extent.width as _,
+                            height: extent.height as _,
+                            depth: 1,
+                        },
+                    ).expect("failed to create framebuffer!"),
+            );
         }
+
+        swapchain_framebuffers
     }
 
-    device.destroy_command_pool(command_pool.into_raw());
+    fn create_command_buffers<'a>(
+        command_pool: &'a mut pool::CommandPool<back::Backend, Graphics>,
+        render_pass: &<back::Backend as Backend>::RenderPass,
+        framebuffers: &[<back::Backend as Backend>::Framebuffer],
+        extent: window::Extent2D,
+        pipeline: &<back::Backend as Backend>::GraphicsPipeline,
+    ) -> Vec<command::Submit<back::Backend, Graphics, command::MultiShot, command::Primary>> {
+        // reserve (allocate memory for) num number of primary command buffers
+        command_pool.reserve(framebuffers.iter().count());
+
+        let mut submission_command_buffers: Vec<
+            command::Submit<back::Backend, Graphics, command::MultiShot, command::Primary>,
+        > = Vec::new();
+
+        for fb in framebuffers.iter() {
+            // command buffer will be returned in 'recording' state
+            // Shot: how many times a command buffer can be submitted; we want MultiShot (allow submission multiple times)
+            // Level: command buffer type (primary or secondary)
+            // allow_pending_resubmit is set to true, as we want to allow simultaneous use per vulkan-tutorial
+            let mut command_buffer: command::CommandBuffer<
+                back::Backend,
+                Graphics,
+                command::MultiShot,
+                command::Primary,
+            > = command_pool.acquire_command_buffer(true);
+
+            command_buffer.bind_graphics_pipeline(pipeline);
+            {
+                // begin render pass
+                let render_area = pso::Rect {
+                    x: 0,
+                    y: 0,
+                    w: extent.width as _,
+                    h: extent.height as _,
+                };
+                let clear_values = vec![command::ClearValue::Color(command::ClearColor::Float([
```

Comment:
Not related to your change, but is this `4.0` supposed to be `1.0`?

Reply:
Yes, it is. I fixed it globally.